src/main/java/io/vertx/core/http/WebSocketBase.java

```diff
@@ -104,6 +105,15 @@
    */
   String closeReason();
 
+  /**
+   *  Returns the HTTP response headers during the websocket connection handler.
+   *  <p/>
+   *  After the completion handler callback has completed the response headers will be {@code null}
+   *
+   * @return the response headers
+   */
+  MultiMap headers();
```

Comment:
we should add this to `WebSocket` instead of `WebSocketBase`

Reply:
> 
> 
> we should add this to `WebSocket` instead of `WebSocketBase`

instead of or both?  If removed form `WebSocketBase`  `WebsocketImplBase` won't have the impl since it is implementing `WebSocketBase` and I don't believe it would compile.